File: dotnet/src/DocGen/Exceptions/DocGenException.cs
namespace Dokmatiq.DocGen.Exceptions;

/// <summary>
/// Base exception for all DocGen SDK errors.
/// </summary>
public class DocGenException : Exception
{
    public DocGenException(string message) : base(message) { }
    public DocGenException(string message, Exception innerException) : base(message, innerException) { }
}

File: java/src/main/java/com/dokmatiq/docgen/exception/DocGenException.java
package com.dokmatiq.docgen.exception;

/** Base exception for all DocGen SDK errors. */
public class DocGenException extends RuntimeException {
    public DocGenException(String message) {
        super(message);
    }

    public DocGenException(String message, Throwable cause) {
        super(message, cause);
    }
}
